app/javascript/components/ansible-playbook-edit-catalog-form/helper.js

```diff
@@ -0,0 +1,315 @@
+import { useFieldApi, useFormApi } from '@@ddf';
+import React from 'react';
+import { Button, TextInput, Checkbox, Select, SelectItem } from 'carbon-components-react';
+import { TrashCan32 } from '@carbon/icons-react';
+import { TreeViewRedux } from '../tree-view';
+
+const convertObject = (inputObj) => {
+  const outputArray = [];
+  for (const [key, value] of Object.entries(inputObj)) {
+    outputArray.push({ key, value: value.default });
+  }
+  return outputArray;
+};
+
+const convertArrayToObject = (arr) => arr.reduce((acc, curr) => {
+  acc[curr.key] = { 'default': curr.value };
+  return acc;
+}, {});
+
+const getTenantId = (key) => {
+  if (key.startsWith('tn')) {
+    return key.split('-')[1];
+  }
+};
+
+const getSortedHash = function (inputHash) {
+  const sortedHash = Object.keys(inputHash)
+    .map((key) => ({ k: key, v: inputHash[key] }))
+    .sort((a, b) => a.v.localeCompare(b.v))
+    .reduce((o, e) => {
+      o[e.k] = e.v;
+      return o;
+    }, {});
+  return sortedHash;
+};
+
+export const getLogOutputTypes = function () {
+  return {
+    on_error: __('On Error'),
+    always: __('Always'),
+    never: __('Never'),
+  };
+};
+
+export const getVerbosityTypes = function () {
+  return {
+    0: '0 (Normal)',
+    1: '1 (Verbose)',
+    2: '2 (More Verbose)',
+    3: '3 (Debug)',
+    4: '4 (Connection Debug)',
+    5: '5 (WinRM Debug)',
+  };
+};
+
+export const formCloudTypes = (data) => {
+  const cloudTypes = {};
+  const embeddedAnsibleCredentialTypes = data.data.credential_types.embedded_ansible_credential_types;
+
+  for (const credType in embeddedAnsibleCredentialTypes) {
+    const credObject = embeddedAnsibleCredentialTypes[credType];
+    if (credObject.type === 'cloud') {
+      cloudTypes[credType] = credObject.label;
+    }
+  }
+  return getSortedHash(cloudTypes);
```

Comment:
```
embeddedAnsibleCredentialTypes.forEach((credType) => {
    const credObject = embeddedAnsibleCredentialTypes[credType];
    if (credObject.type === 'cloud') {
      cloudTypes[credType] = credObject.label;
    }
  });
  return getSortedHash(cloudTypes);
```

Reply:
could you copy-paste the object here?

Reply:
@jeffibm  i have used forEach on object as below
```
 Object.keys(embeddedAnsibleCredentialTypes).forEach((credType) => {
    const credObject = embeddedAnsibleCredentialTypes[credType];
    if (credObject.type === 'cloud') {
      cloudTypes[credType] = credObject.label;
    }
  });
```